gradio/blocks.py

```diff
@@ -618,24 +618,24 @@ def get_block_instance(id: int) -> Block:
                 raise ValueError(f"Cannot find block with id {id}")
             cls = component_or_layout_class(block_config["type"])
 
-            block_config["props"] = cls.recover_kwargs(block_config["props"])
-
             # If a Gradio app B is loaded into a Gradio app A, and B itself loads a
             # Gradio app C, then the proxy_urls of the components in A need to be the
             # URL of C, not B. The else clause below handles this case.
             if block_config["props"].get("proxy_url") is None:
-                block_proxy_url = f"{proxy_url}/"
-            else:
-                block_proxy_url = block_config["props"]["proxy_url"]
-                proxy_urls.add(block_proxy_url)
+                block_config["props"]["proxy_url"] = f"{proxy_url}/"
```

Comment:
Slightly refactored to this code so that `proxy_url` would get passed into `gr.Dataset`